apps/angular/55-back-button-navigation/src/app/simple-action/simple-action.component.ts

```diff
@@ -9,12 +10,26 @@ import { DialogComponent } from '../dialog/dialog.component';
   selector: 'app-simple-action',
   templateUrl: './simple-action.component.html',
 })
-export class SimpleActionComponent {
+export class SimpleActionComponent implements DeactivationCheck {
   readonly #dialog = inject(MatDialog);
+  #dialogRef: MatDialogRef<DialogComponent> | undefined;
+
+  canDeactivate(): boolean {
+    if (this.#dialogRef) {
+      this.#dialogRef.close();
+      return false;
+    }
+    return true;
+  }
 
   openDialog(): void {
-    this.#dialog.open(DialogComponent, {
+    this.#dialogRef = this.#dialog.open(DialogComponent, {
       width: '250px',
+      closeOnNavigation: false,
+    });
+
+    this.#dialogRef.afterClosed().subscribe(() => {
```

Comment:
question: Do afterClosed trigger a complete event ?

Reply:
Good point. I'm not sure.

To be safe, I just added an operator to clean it up.